client/src/pages/Reports.js

```diff
@@ -47,6 +50,73 @@ const Reports = () => {
         admin: "container-reportsTable"
     });
 
+    const createAndDownloadPDF = (ReportId) => {
+        setPdfLoading(true);
+        setCurrentPdfLoading(ReportId);
+
+        let header = {
+            'authorization': "Bearer " + cookies.get("accessToken"),
+        }
+
+        let param = {
+            reportid: ReportId
+        }
+
+        Axios.defaults.withCredentials = true;
+
+        Axios.post(`${process.env.REACT_APP_API}/reports/createPdf`, param, { headers: header })
+            .then((response) => {
+                if(response.data === true) {
+                    Axios.get(`${process.env.REACT_APP_API}/reports/fetchPdf`, { params: param, headers: header, responseType: 'arraybuffer' })
+                    .then((res) => {
+                        const pdfBlob = new Blob([res.data], { type: 'application/pdf' });
+
+                        const url = URL.createObjectURL(pdfBlob);
+
+                        var element = document.createElement('a');
+                        document.body.appendChild(element);
+                        element.style = "display: none";
+                        element.href = url;
+                        element.download = `ChartReport-${ReportId}`;
+                        element.click();
+                        document.body.removeChild(element);
+                        setPdfLoading(false);
+                    })
+                    .catch((error) => {
+                        setPdfLoading(false);
+                        if (error.response) {
+                            if (error.response.status === 403 || error.response.status === 401) {
+                                alert("You are not authorized to perform this action.");
+                            }
+                            else {
+                                alert("Could not fetch pdf file...");
+                            }
+                        }
+                        else if (error.request) {
+                            alert("Could not fetch pdf file...");
+                        }
+                    });
+                }
+                else {
+                    alert("Could not reach b&C Engine...");
+                }
```

Comment:
if `response` comes back as a falsy value(the last else here), it does not mean that the B&C Engine was not reached. However, if the code execution reaches the `else if(error.request)`, that means that the B&C Engine could not be reached.

Reply:
What would you recommend I put instead?

Reply:
in the last else which is highlighted here, I would just put 'Could not fetch Chart Report' or something like that